extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -0,0 +1,230 @@
+		}
+
+		return $user->getId();
+	}
+
+	private function getUserLog( $userName ) {
+		$displayedUserLogRecords = [];
+		$hasNoUserLogRecords = false;
+		$hasUserError = false;
+		$userId = null;
+		$userErrorMessage = null;
+
+		try {
+			$userId = $this->getUserIdByUsername( $userName );
+		} catch ( Exception $e ) {
+			$hasUserError = true;
+		}
+
+		$userLogRecords = $this->aggregateLogSearches( $userId, $userName );
+
+		if ( count( $userLogRecords ) == 0 ) {
+			$hasNoUserLogRecords = true;
+		}
+
+		foreach ( $userLogRecords as $userLogRecord ) {
+			array_push( $displayedUserLogRecords, [
+				'app' => $userLogRecord->getApp(),
+				'ip' => $userLogRecord->getIp(),
+				'language' => $userLogRecord->getLanguage(),
+				'location' => $userLogRecord->getLocation(),
+				'timestamp' => $userLogRecord->getTimestamp(),
+				'userAgent' => $userLogRecord->getUserAgent(),
+			] );
+		}
+
+		return \MustacheService::getInstance()->render(
+			self::$userLogTemplate,
+			[
+				'appHeader' => wfMessage( 'discussionslog-app-header' )
+						->escaped(),
+				'hasUserError' => $hasUserError,
+				'ipAddressHeader' => wfMessage( 'discussionslog-ip-address-header' )
+						->escaped(),
+				'languageHeader' => wfMessage( 'discussionslog-language-header')
+						->escaped(),
+				'locationHeader' => wfMessage( 'discussionslog-location-header' )
+						->escaped(),
+				'logTableCaption' => wfMessage( 'discussionslog-table-caption' )
+						->params( [$userName, $userId] )
+						->escaped(),
+				'userErrorMessage' => wfMessage( 'discussionslog-no-user-match-error' )
+						->escaped(),
+				'hasNoUserLogRecords' => $hasNoUserLogRecords,
+				'noUserLogRecordsMessage' => wfMessage( 'discussionslog-no-mobile-activity-error' )
+						->params( $userName )
+						->escaped(),
+				'timestampHeader' => wfMessage( 'discussionslog-timestamp-header' )
+						->escaped(),
+				'userAgentHeader' => wfMessage( 'discussionslog-user-agent-header' )
+						->escaped(),
+				'userLogRecords' => $displayedUserLogRecords
+			]
+		);
+	}
+
+	private function aggregateLogSearches( $userId, $userName ) {
+		$logger = Wikia\Logger\WikiaLogger::instance();
+		$query = $this->getQuery( $userId );
+		$client = new GuzzleHttp\Client();
+		$records = [];
+		$dayOffset = 0;
+		while ( count( $records ) < self::PAGINATION_SIZE && $dayOffset < self::DAYS_RANGE ) {
+			$url = $this->constructKibanaUrl( $dayOffset++ );
+			try {
+				$response = $client->post(
+					$url, [
+						'body' => $query,
+					]
+				);
+			} catch ( \GuzzleHttp\Exception\RequestException $requestException ) {
+				$logger->warning(
+					sprintf( 'Request to elasticsearch failed: %s', $requestException->getMessage() )
+				);
+				continue;
+			}
+
+			if ( $response->getStatusCode() !== self::HTTP_STATUS_OK ) {
+				$logger->warning(
+					sprintf( 'Elasticsearch request error; status code %d', $response->getStatusCode() )
+				);
+				continue;
```

Comment:
I feel like this line, and L154 should be a `break` rather than a `continue`. If these requests are failing, it doesn't seem like we should hit ES 14 more times.